apps/test/unit/gamelab/GameLabGroupTest.js

```diff
@@ -33,4 +34,237 @@ describe('GameLabGroup', function () {
+        expect(result).to.equal(false);
+      });
+
+      it('returns false for bounceOff', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.bounceOff(targetSprite);
+        }
+        expect(result).to.equal(false);
+      });
+
+      it('returns false for overlap', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.overlap(targetSprite);
+        }
+        expect(result).to.equal(false);
+      });
+
+      it('returns false for displace', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.displace(targetSprite);
+        }
+        expect(result).to.equal(false);
+      });
+    });
+
+    describe('first sprite in group overlaps target sprite', function () {
+      beforeEach(function () {
+        // Make sprite1 overlap with target sprite
+        sprite1.x = targetSprite.x;
+        sprite1.y = targetSprite.y;
+      });
+
+      it('returns true for isTouching', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.isTouching(targetSprite);
+        }
+        expect(result).to.equal(true);
+      });
+
+      it('returns true for collide', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.collide(targetSprite);
+        }
+        expect(result).to.equal(true);
+      });
+
+      it('returns true for bounce', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.bounce(targetSprite);
+        }
+        expect(result).to.equal(true);
+      });
+
+      it('returns true for bounceOff', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.bounceOff(targetSprite);
+        }
+        expect(result).to.equal(true);
+      });
+
+      it('returns true for overlap', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.overlap(targetSprite);
+        }
+        expect(result).to.equal(true);
+      });
+
+      it('returns true for displace', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.displace(targetSprite);
+        }
+        expect(result).to.equal(true);
+      });
+    });
+
+    describe('last sprite in group overlaps target sprite', function () {
+      beforeEach(function () {
+        // Make sprite1 overlap with target sprite
+        sprite2.x = targetSprite.x;
+        sprite2.y = targetSprite.y;
+      });
+
+      it('returns true for isTouching', function () {
+        let result;
+        for (let i = 0; i < group.length + 1; i++) {
+          result = group.isTouching(targetSprite);
+        }
+        expect(result).to.equal(true);
+      });
```

Comment:
These tests all test `group.foo(sprite)`, so it looks like you need to add tests for `sprite.foo(group)` to cover all of the cases that you are fixing.